src/GitVersion.Core/VersionCalculation/TrunkBased/NonTrunk/CommitOnNonTrunkBranchedBase.cs

```diff
@@ -6,18 +6,21 @@ namespace GitVersion.VersionCalculation.TrunkBased.NonTrunk;
 internal abstract class CommitOnNonTrunkBranchedBase : ITrunkBasedIncrementer
 {
     public virtual bool MatchPrecondition(TrunkBasedIteration iteration, TrunkBasedCommit commit, TrunkBasedContext context)
-        => !commit.Configuration.IsMainBranch && commit.BranchName != iteration.BranchName && commit.Successor is null;
+        => !commit.GetEffectiveConfiguration(context.Configuration).IsMainBranch
+           && commit.BranchName != iteration.BranchName
+           && commit.Successor is null;
 
     public virtual IEnumerable<IBaseVersionIncrement> GetIncrements(
         TrunkBasedIteration iteration, TrunkBasedCommit commit, TrunkBasedContext context)
     {
         context.BaseVersionSource = commit.Value;
 
         var incrementForcedByBranch = iteration.Configuration.Increment == IncrementStrategy.Inherit
-            ? commit.GetIncrementForcedByBranch() : iteration.Configuration.Increment.ToVersionField();
+            ? commit.GetIncrementForcedByBranch(context.Configuration) : iteration.Configuration.Increment.ToVersionField();
```

Comment:
Newline here please.

```suggestion
            ? commit.GetIncrementForcedByBranch(context.Configuration)
            : iteration.Configuration.Increment.ToVersionField();
```